executable-spec/src/Cardano/Ledger/Spec/STS/Update/Data.hs

```diff
@@ -1,29 +1,130 @@
+{-# LANGUAGE DeriveAnyClass #-}
 {-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 module Cardano.Ledger.Spec.STS.Update.Data where
 
-import Data.Word (Word64)
+--import Data.Word (Word64)
 import           Data.Monoid.Generic (GenericMonoid (GenericMonoid),
                      GenericSemigroup (GenericSemigroup))
 import           Data.Set (Set)
 import           GHC.Generics (Generic)
-
+import           Data.Text (Text)
+import           Numeric.Natural (Natural)
+import           Data.Hashable (Hashable)
+import qualified Data.Hashable as H
 import qualified Ledger.Core as Core
+import           Data.Map.Strict (Map)
+
+
+-- |A unique ID of a software update
+newtype UpId = UpId { getUpId :: Integer --Word64
```

Comment:
Why is this an `Integer` now?

Reply:
I changed it to an Integer because I was getting a runtime exception (when doing a randomTrace) that (+) has not been defined for Id. Because of the code line (at Ideation.hs: 156)

```
where
                nextId = maximum $ (Data.UpId 0) : fmap ((Data.UpId 1)+) ids
```
I have tried it now again with Word64 and I cannot reproduce the exception. I can change it back to Word64 and test it for a while...

Reply:
I have changed it to Word64.